virtManager/details/viewers.py

```diff
@@ -134,7 +134,7 @@ def _get_fd_for_open(self):
         if not self._vm.conn.support.domain_open_graphics():
             return None
 
-        return self._vm.open_graphics_fd()
+        return self._vm.open_graphics_fd(self._ginfo.gidx)
```

Comment:
All this gidx handling could go into the previous patch. Also kill the idx=0 default param in the first patch, force the one user to make it explicit.

Reply:
I've fixed it. Thanks very much!